Sources/FoundationEssentials/String/String+Path.swift

```diff
@@ -137,6 +137,81 @@ extension String {
+                    fatalError("unable to lookup SID for user \(user)")
+                }
+
+                return withUnsafeTemporaryAllocation(of: CChar.self, capacity: Int(cbSID)) { pSID in
+                    return withUnsafeTemporaryAllocation(of: WCHAR.self, capacity: Int(cchReferencedDomainName)) { pwszReferencedDomainName in
+                        guard LookupAccountNameW(nil, pwszUserName, pSID.baseAddress, &cbSID, pwszReferencedDomainName.baseAddress, &cchReferencedDomainName, &eUse) else {
+                            fatalError("unable to lookup SID for user \(user)")
+                        }
+
+                        var pwszSID: LPWSTR? = nil
+                        guard ConvertSidToStringSidW(pSID.baseAddress, &pwszSID) else {
+                            fatalError("unable to convert SID to string for user \(user)")
+                        }
+
+                        return #"SOFTWARE\Microsoft\Windows NT\CurrentVersion\ProfileList\#\(String(decodingCString: pwszSID!, as: UTF16.self))"#.withCString(encodedAs: UTF16.self) { pwszKeyPath in
+                            return "ProfileImagePath".withCString(encodedAs: UTF16.self) { pwszKey in
+                                var cbData: DWORD = 0
+                                guard RegGetValueW(HKEY_LOCAL_MACHINE, pwszKeyPath, pwszKey, RRF_RT_REG_SZ, nil, nil, &cbData) == ERROR_SUCCESS else {
+                                    fatalError("unable to query ProfileImagePath for user \(user)")
+                                }
+                                return withUnsafeTemporaryAllocation(of: WCHAR.self, capacity: Int(cbData)) { pwszData in
+                                    guard RegGetValueW(HKEY_LOCAL_MACHINE, pwszKeyPath, pwszKey, RRF_RT_REG_SZ, nil, pwszData.baseAddress, &cbData) == ERROR_SUCCESS else {
+                                        fatalError("unable to query ProfileImagePath for user \(user)")
+                                    }
+                                    return String(decodingCString: pwszData.baseAddress!, as: UTF16.self)
+                                }
+                            }
+                        }
+
+                    }
+                }
+            }
+        }
+
+        var hToken: HANDLE? = nil
+        guard OpenProcessToken(GetCurrentProcess(), TOKEN_QUERY, &hToken) else {
+            guard let UserProfile = GetUserProfile() else {
+                fatalError("unable to evaluate `%UserProfile%`")
+            }
+            return UserProfile
+        }
+        defer { CloseHandle(hToken) }
+
+        var dwcchSize: DWORD = 0
+        _ = GetUserProfileDirectoryW(hToken, nil, &dwcchSize)
+
+        return withUnsafeTemporaryAllocation(of: WCHAR.self, capacity: Int(dwcchSize)) {
+            var dwcchSize: DWORD = DWORD($0.count)
+            guard GetUserProfileDirectoryW(hToken, $0.baseAddress, &dwcchSize) else {
+                fatalError("unable to query user profile directory")
```

Comment:
Should we fallback to `USERPROFILE` here too?

Reply:
I didn't fallback here because we just called it above to get the required buffer size. If that succeeded and this failed, I think we are in some weird state and a `fatalError` feels safer.

Reply:
Ah good point thanks for the explanation